src/bench/cluster_linearize.cpp

```diff
@@ -207,6 +210,42 @@ void BenchMergeLinearizationsWorstCase(ClusterIndex ntx, benchmark::Bench& bench
     });
 }
 
+template<size_t N>
+void BenchLinearizeOptimally(benchmark::Bench& bench, const std::array<uint8_t, N>& serialized)
+{
+    // Determine how many transactions the serialized cluster has.
+    ClusterIndex num_tx{0};
+    {
+        SpanReader reader{serialized};
+        DepGraph<BitSet<128>> depgraph;
+        reader >> Using<DepGraphFormatter>(depgraph);
+        num_tx = depgraph.TxCount();
+        assert(num_tx < 128);
+    }
+
+    SpanReader reader{serialized};
+    auto runner_fn = [&]<typename SetType>() noexcept {
+        DepGraph<SetType> depgraph;
+        reader >> Using<DepGraphFormatter>(depgraph);
+        uint64_t rng_seed = 0;
+        bench.run([&] {
+            auto res = Linearize(depgraph, /*max_iterations=*/1000000, rng_seed++);
```

Comment:
is f94c546ac02ad11ba73174cb8d25220515cee893 the first commit that you can assert optimality? I'm not sure having the benchmark in prior to that is so meaningful

Reply:
also seems like the assertion fails for me at that commit